docs/AppFramework.md

```diff
@@ -506,3 +621,88 @@ The App Framework does not preview, analyze, verify versions, or enable Splunk A
+```
+
+3. Get  credentials to access cluster
+```
+az aks get-credentials --resource-group splunkOperatorResourceGroup --name splunkOperatorCluster
+```
+4. Get the Kubelet user managed identity 
+
+Run
+```
+$ az identity list 
+```
+
+Find the section that has <AKS Cluster Name>-agentpool under name
+
+That is look for the block that contains "name": "splunkOperatorCluster-agentpool"
+
+```
+{
+"clientId": "a5890776-24e6-4f5b-9b6c-**************",
+"id": "/subscriptions/f428689e-c379-4712--**************",/resourcegroups/MC_splunkOperatorResourceGroup_splunkOperatorCluster_westus2/providers/Microsoft.ManagedIdentity/userAssignedIdentities/splunkOperatorCluster-agentpool",
+"location": "westus2",
+"name": "splunkOperatorCluster-agentpool",
+"principalId": "f0f04120-6a36-49bc--**************",",
+"resourceGroup": "MC_splunkOperatorResourceGroup_splunkOperatorCluster_westus2",
+"tags": {},
+"tenantId": "8add7810-b62a--**************",",
+"type": "Microsoft.ManagedIdentity/userAssignedIdentities"
+}
+```
+
+Extract the principalId value from the outout above. Or you can use the following command to get the principalId 
+```
+$ az identity show --name <identityName> --resource-group "<resourceGroup>" --query 'principalId' --output tsv
+```
+Example:
+```
+$ principalId=$(az identity show --name splunkOperatorCluster-agentpool --resource-group "MC_splunkOperatorResourceGroup_splunkOperatorCluster_westus2" --query 'principalId' --output tsv)
+$ echo $principalId
+```
+f0f04120-6a36-49bc--**************
+
+5. Assign read access for Kubelet user managed identity to the storage account
+
+Use the `principalId` from the above section and assign it to the storage account
+```
+az role assignment create --assignee "<principalId>" --role 'Storage Blob Data Reader' --scope /subscriptions/<subscription_id>/resourceGroups/<storageAccountResourceGroup>/providers/Microsoft.Storage/storageAccounts/<storageAccountName>
+```
+For my example, if <storageAccountResourceGroup> is splunkOperatorResourceGroup
+
+and <storageAccountName> is mystorageaccount then the command would be:
+```
+$  az role assignment create --assignee "f0f04120-6a36-49bc--**************" --role 'Storage Blob Data Reader' --scope /subscriptions/f428689e-c379-4712--**************/resourceGroups/splunkOperatorResourceGroup/providers/Microsoft.Storage/storageAccounts/mystorageaccount
+```
+After this command, you can use App framework for Azure blob without secrets.
+
+Azure Blob Authorization Recommendations:
+
+Azure allows "Managed Identities" assignment at the "storage accounts" level as well as at specific buckets levels. A managed identity that is assigned read permissions at a storage account level will have read access for all the buckets within that storage account. If you want to restrict the Splunk operator to having access to Splunk apps only from a specific bucket, you can assign the managed identity to that specific bucket only (and not to the whole parent storage account).
```

Comment:
Instead of "If you want to restrict the Splunk Operator" can we change it to something along the lines "We suggest assigning access to specific bucket as a good security practice"

Reply:
Thanks Savan. See how it looks now.

Reply:
@savanp-splunk  I merged develop to this branch so your review was reset. So asking you to review it again. No change to the above content is done, though.